src/Management/src/Endpoint/Loggers/LoggersEndpointMiddleware.cs

```diff
@@ -3,90 +3,103 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Net;
+using System.Text.Json;
 using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
-using Steeltoe.Management.Endpoint.ContentNegotiation;
 using Steeltoe.Management.Endpoint.Middleware;
 using Steeltoe.Management.Endpoint.Options;
 
 namespace Steeltoe.Management.Endpoint.Loggers;
 
-public class LoggersEndpointMiddleware : EndpointMiddleware<Dictionary<string, object>, LoggersChangeRequest>
+internal sealed class LoggersEndpointMiddleware : EndpointMiddleware<ILoggersRequest, Dictionary<string, object>>
 {
-    public LoggersEndpointMiddleware(LoggersEndpoint endpoint, IOptionsMonitor<ManagementEndpointOptions> managementOptions,
-        ILogger<LoggersEndpointMiddleware> logger)
-        : base(endpoint, managementOptions, logger)
+    private readonly IEnumerable<IContextName> _contextNames;
+    private readonly ILogger<LoggersEndpointMiddleware> _logger;
+
+    public LoggersEndpointMiddleware(ILoggersEndpointHandler endpointHandler, IOptionsMonitor<ManagementEndpointOptions> managementOptions,
+        IEnumerable<IContextName> contextNames, ILoggerFactory loggerFactory)
+        : base(endpointHandler, managementOptions, loggerFactory)
     {
+        _contextNames = contextNames;
+        _logger = loggerFactory.CreateLogger<LoggersEndpointMiddleware>();
     }
 
-    public override Task InvokeAsync(HttpContext context, RequestDelegate next)
+    protected override async Task<Dictionary<string, object>> InvokeEndpointHandlerAsync(HttpContext context, CancellationToken cancellationToken)
     {
-        return Endpoint.Options.ShouldInvoke(managementOptions, context, logger) ? HandleLoggersRequestAsync(context) : Task.CompletedTask;
+        ILoggersRequest loggersRequest = await GetLoggersChangeRequestAsync(context);
+
+        if (loggersRequest is ErrorLoggersRequest)
+        {
+            context.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```

Comment:
In response to https://github.com/SteeltoeOSS/Steeltoe/pull/1130#discussion_r1237651522:

> the Loggers endpoint middleware would require the request to be parsed multiple times - I am not sure that is good here.

The diff at https://github.com/SteeltoeOSS/Steeltoe/compare/Separate_middleware_options..fix-loggers-endpoint-statuscode shows how it can be done without reading the response twice.

It also moves the call to `context.HandleContentNegotiation(_logger);` from `EndpointMiddleware.InvokeAsync` into `WriteResponseAsync`, which wasn't addressed yet.

Reply:
Then we have to write a custom converter to account for a change in the shape.

Reply:
I don't think so, because the shape mismatch is handled in the override of `WriteResponseAsync`. I've tried this out, all tests are green.

Reply:
This is done.